# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderBridgeParent.cpp
# Commit: eac17a862c29
# Full Hash: eac17a862c29a52ae1a389c5e55c0a1cec0db7b6
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2020-01-11 09:48:29
# Regressor Bug: 1600472
# File Overlap Count: 2
# Description:
#   Bug 1600472 - Disable allowing sacrificing subpixel anti-aliasing for small screens. r=jrmuizel
#   
#   This patch only allows sacrificing subpixel anti-aliasing when the
#   screen size is larger than WUXGA, and when the force disable pref is not
#   set. In the future, we may also add disable this for high end GPUs.
# ==============================================================================

diff -r b6e5faa0693d -r eac17a862c29 gfx/layers/wr/WebRenderBridgeParent.cpp
--- a/gfx/layers/wr/WebRenderBridgeParent.cpp	Fri Jan 10 18:24:13 2020 +0000
+++ b/gfx/layers/wr/WebRenderBridgeParent.cpp	Fri Jan 10 16:57:55 2020 +0000
@@ -16,6 +16,7 @@
 #include "mozilla/Range.h"
 #include "mozilla/StaticPrefs_gfx.h"
 #include "mozilla/UniquePtr.h"
+#include "mozilla/gfx/gfxVars.h"
 #include "mozilla/layers/AnimationHelper.h"
 #include "mozilla/layers/APZSampler.h"
 #include "mozilla/layers/APZUpdater.h"
@@ -358,6 +359,9 @@
     MOZ_ASSERT(api);
     mApis[api->GetRenderRoot()] = api;
   }
+
+  UpdateDebugFlags();
+  UpdateQualitySettings();
 }
 
 WebRenderBridgeParent::WebRenderBridgeParent(const wr::PipelineId& aPipelineId)
@@ -1728,6 +1732,26 @@
   mApis[wr::RenderRoot::Default]->WaitFlushed();
 }
 
+void WebRenderBridgeParent::UpdateQualitySettings() {
+  for (auto& api : mApis) {
+    if (!api) {
+      continue;
+    }
+    wr::TransactionBuilder txn;
+    txn.UpdateQualitySettings(gfxVars::AllowSacrificingSubpixelAA());
+    api->SendTransaction(txn);
+  }
+}
+
+void WebRenderBridgeParent::UpdateDebugFlags() {
+  for (auto& api : mApis) {
+    if (!api) {
+      continue;
+    }
+    api->UpdateDebugFlags(gfxVars::WebRenderDebugFlags());
+  }
+}
+
 #if defined(MOZ_WIDGET_ANDROID)
 void WebRenderBridgeParent::RequestScreenPixels(
     UiCompositorControllerParent* aController) {